master/lustrefactory.py

```diff
@@ -72,6 +83,54 @@ def buildCommand(props):
 
     return args
 
+def getTarballFactory(gerrit_repo, **kwargs):
+    """ Generates a build factory for a tarball generating builder.
+    Args:
+        gerrit_repo (string): Gerrit repo url
+    Returns:
+        BuildFactory: Build factory with steps for generating tarballs.
+    """
+    bf = util.BuildFactory()
+
+    # Pull the patch from Gerrit
+    bf.addStep(Gerrit(
+        repourl=gerrit_repo, workdir="build/lustre",
+        mode="full", method="fresh", retry=[60,60], timeout=3600,
+        logEnviron=False, getDescription=True, haltOnFailure=True,
+        description=["cloning"], descriptionDone=["cloned"]))
+
+    # make tarball
+    bf.addStep(ShellCommand(
+        command=['sh', './autogen.sh'],
+        description=["autogen"],
+        descriptionDone=["autogen"],
+        workdir="build/lustre"))
+
+    bf.addStep(Configure(
+        command=['./configure', '--enable-dist'],
+        workdir="build/lustre"))
+
+    bf.addStep(Compile(
+        command=['make', 'dist'],
+        workdir="build/lustre"))
+
+    # upload it to the master
+    bf.addStep(SetPropertyFromCommand(
+        command=['sh', '-c', 'ls *.tar.gz'],
```

Comment:
That will work, but the _ls_ is redundant.  You are doing a directory lookup in the shell with the asterisk, and then a second lookup in _ls_.  It would be simpler to use _echo_.